### PR TITLE
Ticket model & TicketDAO

### PR DESCRIPTION
**Pull request by Anubhavhumber**
PR includes:
Ticket Model & DAO + support menu item.

### CODE REVIEW

**Review by zahnno — APPROVED** [review]

File: src/foam/support/model/Ticket.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
```
Comment (kgrgreer) [inline src/foam/support/model/Ticket.js]: 2018

File: src/foam/support/model/Ticket.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      class: 'String',
+      name: 'subject'
+    },
+    {
+      class: 'String',
+      name: 'publicMessage'
+    },
+    {
+      class: 'DateTime',
+      name: 'time'
```
Comment (kgrgreer) [inline src/foam/support/model/Ticket.js]: What time is this? LastModified?

File: src/services [hunk 3]
```diff
@@ -53,3 +53,5 @@ p({"class":"foam.nanos.boot.NSpec", "name":"professorDAO", "lazy":true, "serve":
 p({"class":"foam.nanos.boot.NSpec", "name":"studentDAO", "lazy":true, "serve":true, "serviceScript":"return new foam.dao.EasyDAO.Builder(x).setJournaled(true).setJournalName(\"students\").setOf(foam.nanos.demo.relationship.Student.getOwnClassInfo()).build();", "client":"{\"of\":\"foam.nanos.demo.relationship.Student\", \"remoteListenerSupport\": true}"})
 p({"class":"foam.nanos.boot.NSpec", "name":"courseDAO", "lazy":true, "serve":true, "serviceScript":"return new foam.dao.EasyDAO.Builder(x).setJournaled(true).setJournalName(\"courses\").setGuid(true).setOf(foam.nanos.demo.relationship.Course.getOwnClassInfo()).build();", "client":"{\"of\":\"foam.nanos.demo.relationship.Course\", \"remoteListenerSupport\": true}"})
 p({"class":"foam.nanos.boot.NSpec", "name":"studentCourseJunctionDAO", "lazy":true, "serve":true, "serviceScript":"return new foam.dao.EasyDAO.Builder(x).setJournaled(true).setJournalName(\"studentCourseJunction\").setOf(foam.nanos.demo.relationship.StudentCourseJunction.getOwnClassInfo()).build();", "client":"{\"of\":\"foam.nanos.demo.relationship.StudentCourseJunction\", \"remoteListenerSupport\": true}"})
+
```
Comment (kgrgreer) [inline src/services:56]: Put in its own services file under support.

Comment (zahnno) [reply src/services:56]: ignore comment

**Comment by Anubhavhumber** [comment]
Ok Kevin...

File: src/foam/support/model/Ticket.js (hunk 2 above)
Comment (zahnno) [reply src/foam/support/model/Ticket.js]: remove this property

File: src/foam/support/support.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 FOAM_FILES([
   //Place support files here.
   // example: { name: 'foam/support/Test' }
+  { name: "foam/support/model/Ticket" }
```
Comment (adamvy) [inline src/foam/support/support.js]: Does this really need its own FOAM_FILES?  Can we just add it to the nano.js files?

Especially if we're adding it to the menus, we should make sure the models are loaded when nanos is.

File: src/permissions [hunk 5]
```diff
@@ -1,4 +1,4 @@
 p({"class":"foam.nanos.auth.Permission", "id":"*",         "description":"Do anything global permission." })
 p({"class":"foam.nanos.auth.Permission", "id":"menu.auth", "description":"Perform authentication related configuration" })
 p({"class":"foam.nanos.auth.Permission", "id":"service.*", "description":"Global permission for reading of services"})
-p({"class":"foam.nanos.auth.Permission", "id":"service.read.*", "description":"Global permission for reading of services from a DAO"})
```
Comment (adamvy) [inline src/permissions:4]: Why did this file get changed?

File: src/scripts [hunk 6]
```diff
@@ -2,4 +2,4 @@ p({"class":"foam.nanos.script.Script","id":"279885d4-28ad-4af8-b1d9-69e6105970e1
...
 p({"class":"foam.nanos.script.Script","id":"showJDAODirectory","description":"Discover where JDAO files are being saved.","lastRun":"2018-01-10T17:09:54.779Z","server":true,"scheduled":false,"code":"import foam.dao.*;\n\nsetAccessibility(true);\n\nDAO dao = x.get(\"userDAO\");\n\nwhile ( ! ( dao instanceof JDAO ) ) dao = dao.getDelegate();\n\nprint(dao.file_);","output":"/Users/kgr/NANOPAY/users\n","notes":""})
-p({"class":"foam.nanos.script.Script","id":"benchmarkHashing","description":"Concurrent Hashing Benchmark","code":"import foam.nanos.bench.*;\nimport net.nanopay.bench.*;\n\nx = x.put(\"pmLogger\", null);\nb = new HashingBenchmark();\nbuilder = new BenchmarkRunner.Builder(x).setThreadCount(2).setInvocationCount(10000).setBenchmark(b);\nrunner = builder.build();\nrunner.execute(x);"})
```
Comment (adamvy) [inline src/scripts:5]: Ditto.  I think you need to configure your editor to put newlines at the end of files.

File: src/users [hunk 7]
```diff
@@ -1 +1 @@
...
..."},"type":"","birthday":"1982-07-07T23:12:00.0Z","address":{"class":"foam.nanos.auth.Address","type":"","verified":false,"deleted":false,"structured":true,"address1":"","address2":"","suite":"300","city":"Toronto","postalCode":"M6G2H3","countryId":"CA","regionId":"ON","encrypted":false,"latitude":43.68018341064453,"longitude":-79.4206314086914,"streetNumber":"","streetName":"","hours":[{"class":"foam.nanos.auth.Hours","day":0,"open":true},{"class":"foam.nanos.auth.Hours","day":1,"open":true},{"class":"foam.nanos.auth.Hours","day":2,"open":true},{"class":"foam.nanos.auth.Hours","day":3,"open":true},{"class":"foam.nanos.auth.Hours","day":4,"open":true},{"class":"foam.nanos.auth.Hours","day":5,"open":true},{"class":"foam.nanos.auth.Hours","day":6,"open":true}]},"language":"en","timeZone":"","password":"SEDjby8u5eM=:ION/k78lY9RJx74MyzbWvfAtPGxoL6NRIQ6zXKZJbqHGHJ4LP6XEELayaUqDxfkOeZ8135M4pG6M+KadqqaJ/w==","previousPassword":"","note":"","businessName":"","businessIdentificationNumber":"","issuingAuthority":"","bankIdentificationCode":"","website":"","group":"system","deviceToken":"","businessTypeId":0,"businessSectorId":1,"branchId":"","clearingId":"","status":"","emailVerified":false})
```
Comment (adamvy) [inline src/users:1]: Ditto

**Comment by Anubhavhumber** [comment]
I signed it!

**Pull request closed without merging**